securedrop_client/api_jobs/base.py

```diff
@@ -91,3 +91,23 @@ def call_api(self, api_client: API, session: Session) -> Any:
         failure.
         '''
         raise NotImplementedError
+
+
+class SingleObjectApiJob(ApiJob):
+    def __init__(self, uuid: str, remaining_attempts: int = DEFAULT_NUM_ATTEMPTS) -> None:
+        super().__init__(remaining_attempts)
+        '''
+        UUID of the item (source, reply, submission, etc.) that this item
+        corresponds to. We track this to prevent the addition of duplicate jobs.
+        '''
+        self.uuid = uuid
+
+    def __repr__(self) -> str:
+        return "{}('{}', {})".format(self.__class__.__name__, self.uuid, self.remaining_attempts)
+
+    def __eq__(self, other: Any) -> bool:  # type: ignore[override]
+        # https://github.com/python/mypy/issues/2783
+        if self.uuid == getattr(other, 'uuid', None) and type(self) == type(other):
```

Comment:
somehow i missed that you were already checking type! this resolves my comment from earlier today.